ledger-api/grpc-definitions/com/daml/ledger/api/v1/active_contracts_service.proto

```diff
@@ -56,9 +58,59 @@ message GetActiveContractsResponse {
 
   // The workflow that created the contracts.
   // Must be a valid LedgerString (as described in ``value.proto``).
+  // Optional, In the Ledger API 3.0 this field will be removed.
   string workflow_id = 2;
 
   // The list of contracts that were introduced by the workflow with ``workflow_id`` at the offset.
   // Must be a valid LedgerString (as described in ``value.proto``).
+  // In the Ledger API 3.0 this field will be removed.
+  // This is empty for multi-domain API.
   repeated CreatedEvent active_contracts = 3;
+
+  // Non-empty for multi-domain API.
+  repeated ContractState contract_states = 4;
+}
+
+// In the entire stream of ContractState-s there can be multiple ContractState messages belong to the same contract.
+message ContractState {
+  oneof contract_state {
+    // The contract is active in the given domain. Only one such message is expected per domain.
+    ActiveContract active_contract = 1;
+
+    // This message is only available, if the corresponding transfer-in event cannot be found before or at the
+    // active_at_offset, and for the querying parties it is possible to observe this event after the active_at_offset,
+    // and also to submit the corresponding transfer-in command.
+    // This event typically means, that the transfer is in-flight.
+    IncompleteTransferredOut incomplete_transferred_out = 2;
+
+    // Important: this message is not indicating that the contract is active for the target domain!
+    // This message is only available, if the corresponding transfer-out event cannot be found before or at the
+    // active_at_offset, and for the querying parties it is possible to observe this event after the active_at_offset.
```

Comment:
there seems to be a distinction here where for incomplete transfer ins you say it's sufficient that I can observe it whereas for incomplete transfer outs you have the additional requirement that I can submit the transfer in. Is that accurate? If my participant is not transferring could I not still see both the transfer in and transfer out but we wouldn't include it as incomplete here?

Reply:
as discussed offline both  incomplete event need to be based on the transferring property